### ISSUE TITLE
Ability to turn off registration

### ISSUE THREAD

**Issue by ebauman** [boxboat/hobbyfarm-chart#303]
**What would you like to be added**: A setting to turn off registration

**Why is this needed**: Folks can register in instances where we may not want them to - perhaps we provision their accounts ahead of time.

This is probably also good to have when improvements are made to auth systems - we won't want to register non-auth-associated users.

**Comment by ebauman** [comment]
I think this probably calls for the creation of `settings.hobbyfarm.io`. Stealing the logic here from Rancher a bit..

```
apiVersion: v1
kind: Setting
metadata:
  name: registration-enabled
displayName: Registration Enabled
value: "false"
```

**Comment by ebauman** [comment]
It's better to do it this way as opposed to a config CR because then you don't have to update the CRD every time you add a new setting.

It's also easier to do it this way for UI purposes as you don't have to parse a configmap full of JSON or YAML. Users can also modify Hf behavior easily by `kubectl edit setting registration-enabled` versus a block of config text.

**Comment by jggoebel** [comment]
This was implemented in HobbyFarm 3.0
https://github.com/hobbyfarm/gargantua/pull/146 + https://github.com/hobbyfarm/ui/pull/168
